library/src/main/java/com/hannesdorfmann/adapterdelegates4/AdapterDelegate.java

```diff
@@ -45,6 +47,25 @@
      */
     protected abstract boolean isForViewType(@NonNull T items, int position);
 
+    /**
+     * Return the view type of the item at <code>position</code> for the purposes
+     * of view recycling.
+     *
+     * <p>The default implementation of this method returns 0, making the assumption of
+     * a single view type for the adapter. Unlike ListView adapters, types need not
+     * be contiguous. Consider using id resources to uniquely identify item view types.
+     *
+     * see detail {@link AdapterDelegatesManager#getItemViewType(T items, int position)}
+     *
+     * @param items The data source of the Adapter
+     * @param position The position in the datasource
+     * @return integer value identifying the type of the view needed to represent the item at
+     *                 <code>position</code>. Type codes need not be contiguous.
+     */
+    protected int getItemViewType(T items, int position) {
```

Comment:
what is this good for? i don't fully understand why we need this? which problem does it solve?

Reply:
This is for nested recyclerviews with shared view pool.

Reply:
Need a bit more time to think about it.

Reply:
I would propose the following: 
Let's not introduce this right now. Let's first introduce the ViewBinding DSL and then as a next step discuss how this change can be introduced by keeping a clean API. 

Can you please remove this from this PR?